scripts/syncLang.py

```diff
@@ -1,277 +1,211 @@
+        :param command: a shell command
+        :return: the output of the shell command
+        """
+        return subprocess.check_output(command.split(" ")).decode("utf-8").rstrip()
+
+
+class Keys:
+    def __init__(self, lines):
+        self.lines = lines
+
+    def duplicates(self) -> list:
+        """
+        return: list of unicode strings
+        """
+        duplicates = []
+        keys_checked = {}
+        for line in self.lines:
+            key, value = self.__extract_key_and_value(line=line)
+            if key:
+                if key in keys_checked:
+                    duplicates.append("{key}={value}".format(key=key, value=value))
+                    translation_in_list = "{key}={value}".format(key=key, value=keys_checked[key])
+                    if translation_in_list not in duplicates:
+                        duplicates.append(translation_in_list)
+                else:
+                    keys_checked[key] = value
+        return duplicates
+
+    def fix_duplicates(self) -> tuple:
+        """
+        Fixes all unambiguous duplicates
+        :return: (list of unicode strings, list of unicode strings): not fixed ambiguous duplicates, fixed unambiguous duplicates
+        """
+        keys = {}
+        fixed = []
+        not_fixed = []
+        for line in self.lines:
+            key, value = self.__extract_key_and_value(line=line)
+            if key:
+                if key in keys:
+                    if not keys[key]:
+                        fixed.append("{key}={value}".format(key=key, value=keys[key]))
+                        keys[key] = value
+                    elif not value:
+                        fixed.append("{key}={value}".format(key=key, value=value))
+                    elif keys[key] == value:
+                        fixed.append("{key}={value}".format(key=key, value=value))
+                    elif keys[key] != value:
+                        not_fixed.append("{key}={value}".format(key=key, value=value))
+                        not_fixed.append("{key}={value}".format(key=key, value=keys[key]))
+                else:
+                    keys[key] = value
 
+        return keys, not_fixed, fixed
+
+    def keys_from_lines(self) -> list:
+        """
+        Builds a list of all translation keys in the list of lines.
+
+        :return: list of unicode strings: the sorted keys within the lines
+        """
+        keys = list()
+        for line in self.lines:
+            key = self.key_from_line(line)
+            if key:
+                keys.append(key)
+        return keys
+
+    @staticmethod
+    def key_from_line(line) -> str:
+        """
+        Tries to extract the key from the line
+
+        :param line: unicode string
+        :return: unicode string: the key or None
+        """
+        if line.find("#") != 0 or line.find("!") != 0:
+            index_key_end = line.find("=")
+            while (index_key_end > 0) and (line[index_key_end - 1] == "\\"):
```

Comment:
Just curious: Is this platform independent line endings?

Reply:
Yep, I have enforced the line ending handling.